air-quality-backend/src/air_quality/etl/forecast/forecast_adapter.py

```diff
@@ -107,3 +113,309 @@ def transform(forecast_data: ForecastData, locations: list[AirQualityLocation])
+        ).strftime("%Y-%m-%d %H:%M:%S")
+        time_steps_dict[len(chunks) - 1] = {
+            "time_start": time_start,
+            "time_end": time_end,
+        }
+
+    return chunks, time_steps_dict
+
+
+def _create_output_directory(forecast_data: ForecastData) -> Tuple[str, str]:
+    """
+    Create output directory for data textures.
+
+    :param forecast_data: The forecast data object.
+
+    :return: The output directory path and forecast date.
+    """
+
+    forecast_date = datetime.fromtimestamp(
+        forecast_data.get_time_value(), timezone.utc
+    ).strftime("%Y-%m-%d_%H")
+
+    output_directory = f"/app/data_textures/{forecast_date}"
+    if not os.path.exists(output_directory):
+        output_directory = f"{os.getcwd()}/data_textures/{forecast_date}"
+    os.makedirs(output_directory, exist_ok=True)
+    return output_directory, forecast_date
+
+
+def _save_data_texture(
+    rgb_data_array: np.ndarray,
+    output_directory: str,
+    forecast_date: str,
+    variable: str,
+    num_chunk: str,
+    total_chunks: str,
+    file_format: str,
+) -> str:
+    """
+    Save data textures to disk and convert to binary data.
+
+    :param rgb_data_array: Numpy array containing the image data.
+    :param output_directory: Directory to save the textures.
+    :param forecast_date: Forecast date string.
+    :param variable: Variable name to include in the filename.
+    :param num_chunk: Chunk number.
+    :param total_chunks: Total number of chunks.
+    :param file_format: File format for saving the texture.
+
+    :return: The path to the saved file.
+    """
+    if variable == "winds_10m":
+        image_array = rgb_data_array
+        format = "RGB"
+    else:
+        image_array = rgb_data_array.squeeze(axis=2)
+        format = "L"
+
+    image = Image.fromarray(image_array, format)
+    output_file = (
+        f"{output_directory}/{variable}_{forecast_date}_CAMS_global."
+        f"chunk_{num_chunk}_of_{total_chunks}.{file_format}"
+    )
+    image.save(output_file, format=file_format, lossless=True)
+    return output_file
+
+
+def _process_variable(
+    data: xr.Dataset,
+    data_type: str,
+    variable_name: str,
+    output_directory: str,
+    forecast_date: str,
+) -> List[Dict[str, str]]:
+    """
+    Process a specific variable and create data textures.
+
+    :param data: The dataset containing the variable.
+    :param data_type: The type of the data.
+    :param variable_name: The name of the variable.
+    :param output_directory: Directory to save the textures.
+    :param forecast_date: The forecast date.
+
+    :return: A list of dictionaries containing data texture metadata for database.
+    """
+    documents = []
+    chunks_per_texture = 16
+    file_format = "webp"
+
+    rgb_data_array, min_value, max_value, units, num_lon, time_vector = _convert_data(
+        data, data_type
+    )
+    chunk_list, chunk_dict = _chunk_data_array(
```

Comment:
I'm wondering if we should put all of this code, including chunk_data_array and save_data_texture into a separate forecast_texture_storer file, so that all of the 'texture storage' parts are separated off from here. 

This file can then just concentrate on the data conversion, and the other file can encapsulate how best to store it.

Reply:
see above, process_variable now only calls convert_data and save_data_tetxures